README.md

```diff
@@ -22,82 +22,201 @@ yarn add -S react-plaid-link
 
-```jsx
-import React, { useCallback } from 'react';
-import { usePlaidLink } from 'react-plaid-link';
+ℹ️ Note: `link_token` cannot be null when passed to `usePlaidLink`. Generate your link_token
+outside of the component where the hook is initialized.
 
-const App = () => {
-  const onSuccess = useCallback((token, metadata) => {
-    // send token to server
-  }, []);
+```tsx
+import React, { useCallback, useState, FunctionComponent } from "react";
+import {
+  usePlaidLink,
+  PlaidLinkOptions,
+  PlaidLinkOnSuccess,
+} from "react-plaid-link";
+
+interface Props {
+  token: string;
+}
 
-  const config = {
-    token: '<GENERATED_LINK_TOKEN>',
+const PlaidLink: FunctionComponent<Props> = ({ token }) => {
+  const onSuccess = useCallback<PlaidLinkOnSuccess>(
+    (public_token, metadata) => {
+      // send public_token to server
+    },
+    []
+  );
+
+  const config: PlaidLinkOptions = {
+    token,
     onSuccess,
-    // ...
+    // onExit
+    // onEvent
   };
 
   const { open, ready, error } = usePlaidLink(config);
 
   return (
-    <MyButton onClick={() => open()} disabled={!ready}>
+    <button onClick={() => open()} disabled={!ready}>
       Connect a bank account
-    </MyButton>
+    </button>
+  );
+};
+
+const App: FunctionComponent = () => {
+  const [token, setToken] = useState<string | null>(null);
+
+  // generate a link_token
+  React.useEffect(() => {
+    async function createLinkToken() {
+      let response = await fetch("/api/create_link_token");
+      const { link_token } = await response.json();
+      setToken(link_token);
+    }
+    createLinkToken();
+  }, []);
+
+  // only initialize Link once our token exists
+  return token === null ? (
+    <div className="loader"></div>
+  ) : (
+    <PlaidLink token={token} />
   );
 };
+
 export default App;
 ```
 
-### Using a React component
+#### OAuth / opening Link without a button click
+
+Handling OAuth redirects requires opening link without any user input. This can
+also be useful if usimply if you want Link to open immediately when your page
+or component renders.
+
+```tsx
+import React, { useCallback, useEffect, FunctionComponent } from "react";
+import {
+  usePlaidLink,
+  PlaidLinkOptions,
+  PlaidLinkOnSuccess,
+} from "react-plaid-link";
 
-```jsx
-import React from 'react';
-import { PlaidLink } from 'react-plaid-link';
+interface Props {
+  token: string;
+}
 
-const App = props => {
-  const onSuccess = (token, metadata) => {
-    // send token to server
+const OpenPlaidLink: FunctionComponent<Props> = ({ token }) => {
+  const onSuccess = useCallback<PlaidLinkOnSuccess>(
+    (public_token, metadata) => {
+      // send public_token to server
+    },
+    []
+  );
+
+  const config: PlaidLinkOptions = {
+    // When re-initializing Link after OAuth redirection, the same
+    // Link token from the first initialization must be used
+    token,
+    onSuccess,
+    // receivedRedirectUri: document.location.href, // required for OAuth
```

Comment:
is the `// required for OAuth` supposed to be on a new line?

Reply:
no, its designed so they can uncomment this line and `// required for OAuth` remains a comment